pkg/operation/care/health.go

```diff
@@ -391,6 +431,43 @@ func (h *Health) checkSystemComponents(
 	return &c, nil
 }
 
+func (h *Health) checkSeedSystemComponents(
+	ctx context.Context,
+	checker *HealthChecker,
+	condition gardencorev1beta1.Condition,
+) (*gardencorev1beta1.Condition,
+	error) {
+
+	managedResources := managedResourcesSeed.List()
+
+	for _, name := range managedResources {
+		mr := &resourcesv1alpha1.ManagedResource{}
+		if err := h.seedClient.Client().Get(ctx, kutil.Key(v1beta1constants.GardenNamespace, name), mr); err != nil {
+			return nil, err
+		}
+
+		if exitCondition := checker.CheckManagedResource(condition, mr); exitCondition != nil {
+			return exitCondition, nil
+		}
+	}
+
+	deployments := deploymentsSeed.List()
+
+	for _, namespace := range deployments {
```

Comment:
This is highly confusing (the `deployments` does not contain deployment names but rather namespace names) 👀 
Can we drop all of this for now? As suggested in #5154, we can make it a bit easier and only check `ManagedResource`s in the garden namespace. As #2754 strives to move everything to `ManagedResource`s, this code here will become obsolete eventually anyways.

Reply:
This is for the `Istio`. Basically we get all deployments for the two namespaces. I was not sure if we were going to add managed resource for them so I did this.

Reply:
We will since this will also allow us to get rid of such "duplicate" health checking code (GRM is already doing it for us).